sentinel/test/unit/due_tasks.js

```diff
@@ -204,3 +218,109 @@ exports['set all due scheduled tasks to pending and handles nonrepeated rows'] =
   });
 
 };
+
+exports['generates the messages for all due scheduled tasks'] = test => {
+  const due = moment().toISOString();
+  const notDue = moment().add(7, 'days').toISOString();
+  const id = 'xyz';
+  const expectedPhone = '5556918';
+  const translate = sinon.stub(utils, 'translate').returns('Please visit {{patient_name}} asap');
+  const getRegistrations = sinon.stub(utils, 'getRegistrations').callsArgWith(1, null, []);
+  const getPatientContact = sinon.stub(utils, 'getPatientContact').callsArgWith(2, null, { name: 'jim' });
+
+  const db = {
+    view: () => {}
+  };
+  const minified = {
+    fields: {
+      patient_id: '123'
+    },
+    contact: {
+      _id: 'a',
+      parent: {
+        _id: 'b'
+      }
+    },
+    scheduled_tasks: [
+      {
+        due: due,
+        state: 'scheduled',
+        message_key: 'visit-1',
+        recipient: 'clinic'
+      },
+      {
+        due: notDue,
+        state: 'scheduled',
+        message_key: 'visit-1',
+        recipient: 'clinic'
+      }
+    ]
+  };
+  const hydrated = {
+    fields: {
+      patient_id: '123'
+    },
+    contact: {
+      _id: 'a',
+      type: 'person',
+      parent: {
+        _id: 'b',
+        type: 'clinic',
+        contact: {
+          _id: 'c',
+          type: 'person',
+          phone: expectedPhone
+        }
+      }
+    },
+    scheduled_tasks: [
+      {
+        due: due,
+        state: 'scheduled',
+        message_key: 'visit-1',
+        recipient: 'clinic'
+      },
+      {
+        due: notDue,
+        state: 'scheduled',
+        message_key: 'visit-1',
+        recipient: 'clinic'
+      }
+    ]
+  };
+  const view = sinon.stub(db, 'view').callsArgWith(3, null, {
+    rows: [
+      {
+        id: id,
+        key: due,
+        doc: minified
+      }
+    ]
+  });
+  sinon.stub(lineage, 'hydrateDocs').returns(Promise.resolve([ hydrated ]));
```

Comment:
If this is stubbed, why bother generating something that really looks hydrated?  Couldn't you just return `const HYDRATED_PLACEHOLDER = {};`?

Reply:
This test checks the logic of what happens when a doc is returned so you have to return a doc or you're not actually testing anything. For example, one line in the unit is

```js
doc.scheduled_tasks.forEach(task => {...
```

So you have to return a doc that defines an array of `scheduled_tasks` or you won't actually test that logic at all.